cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -833,10 +837,11 @@ private SearchResults searchByCustomIndex(SearchRequest request) throws BadReque
     Set<EntityScope> entityScopes = searchRequest.getEntityScopes();
     List<SearchTerm> searchTerms = new LinkedList<>();
     Consumer<String> termAdder = determineSearchFields(namespace, entityScopes, searchTerms);
-    String searchQuery = searchRequest.getQuery();
-    for (String term : Splitter.on(SPACE_SEPARATOR_PATTERN).omitEmptyStrings().trimResults().split(searchQuery)) {
-      termAdder.accept(term);
+
+    for (QueryTerm q : QueryParser.parse(searchRequest.getQuery())) {
```

Comment:
This can be done in Java Stream style too:

```java
QueryParser.parse(searchRequest.getQuery()).forEach(termAdder);
```

Reply:
In this case what is the termAdder?

Reply:
never mind, got it and implemented the stream